components/threading/global_threads_link/global_threads_link.tsx

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {Link, useRouteMatch, useLocation, matchPath} from 'react-router-dom';
+import classNames from 'classnames';
+import {useIntl} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {getThreadCountsInCurrentTeam} from 'mattermost-redux/selectors/entities/threads';
+import {isCollapsedThreadsEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {getThreads} from 'mattermost-redux/actions/threads';
+
+import {t} from 'utils/i18n';
+
+import {isUnreadFilterEnabled} from 'selectors/views/channel_sidebar';
+import {useThreadRouting} from '../hooks';
+
+import ChannelMentionBadge from 'components/sidebar/sidebar_channel/channel_mention_badge';
+
+import ThreadsIcon from './threads_icon';
+
+import './global_threads_link.scss';
+
+const GlobalThreadsLink = () => {
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+    const isFeatureEnabled = useSelector(isCollapsedThreadsEnabled);
+
+    const {url} = useRouteMatch();
+    const {pathname} = useLocation();
+    const inGlobalThreads = matchPath(pathname, {path: '/:team/threads/:threadIdentifier?'}) != null;
+    const {currentTeamId, currentUserId} = useThreadRouting();
+
+    const counts = useSelector(getThreadCountsInCurrentTeam);
+    const unreadsOnly = useSelector(isUnreadFilterEnabled);
+    const someUnreadThreads = counts?.total_unread_threads;
+
+    useEffect(() => {
+        // load counts if necessary
+        if (isFeatureEnabled) {
+            dispatch(getThreads(currentUserId, currentTeamId, {perPage: 5}));
+        }
+    }, [currentTeamId]);
+
+    if (!isFeatureEnabled || (unreadsOnly && !inGlobalThreads && !someUnreadThreads)) {
+        // hide link if feature disabled or filtering unreads and there are no unread threads
+        return null;
+    }
```

Comment:
nit: shouldn't this be before the `useEffect` call?

Reply:
No, hooks need to be called in the same order every time (and before early returns)